core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -554,7 +554,7 @@ class TileMap {
         tileList.forEach {
             for (unit in it.getUnits()) if (unit.owner == player.chosenCiv) unit.removeFromTile()
         }
-        startingLocations.removeAll(startingLocations.filter { it.nation == player.chosenCiv }) // filter creates a copy, no concurrent modification
+        startingLocations.removeAll { it.nation == player.chosenCiv } // filter creates a copy, no concurrent modification
```

Comment:
Is this comment still up-to-date after removing the `filter`?

Reply:
Probably, no. Will remove it as well.

Reply:
It seems to imply that a concurrent modification exception would occur without the filter? Are we sure this wouldn't happen after removing the filter?

Reply:
See my comments below:

> My assumption is that `removeAll` is thread-safe and it will not cause the problem removing own items by given criteria without a need of making the copy, otherwise that's a serious bug in the Kotlin collections.

Reply:
... but it's possible and can be hoped for that removeAll(predicate) has its own mechanism. Jack is right, the special case removeAll is running with a lambda and should know side effects are possible. But I would still check up on it and comment. A look at Kdoc might be enough ...